testdata/project-v3-with-deploy-image/controllers/busybox_controller.go

```diff
@@ -120,6 +127,35 @@ func (r *BusyboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// the desired state on the cluster
 		return ctrl.Result{Requeue: true}, nil
 	}
+
+	// Check if the Busybox instance is marked to be deleted, which is
+	// indicated by the deletion timestamp being set.
+	isBusyboxMarkedToBeDeleted := busybox.GetDeletionTimestamp() != nil
+	if isBusyboxMarkedToBeDeleted {
+		if controllerutil.ContainsFinalizer(busybox, busyboxFinalizer) {
+			// Run finalization logic for busyboxFinalizer. If the
+			// finalization logic fails, don't remove the finalizer so
+			// that we can retry during the next reconciliation.
+			r.recorder.Event(busybox, "Normal", "Deleting", fmt.Sprintf("Custom Resource Definition %s/%s is being deleted", busybox.Namespace, busybox.Name))
+			r.recorder.Event(found, "Normal", "Deleting", fmt.Sprintf("Deployment %s/%s is being deleted", busybox.Namespace, busybox.Name))
```

Comment:
@camilamacedo86 Could you please take a look at these changes for creating an event.
What are we expecting this finalizer to do? Do we expect to log the information we record in the event when we delete the crd?

But by these changes when I delete the crd created the implementation of kubectl delete crd ** stops at just: `customresourcedefinition.apiextensions.k8s.io "memcacheds.example.com.my.domain" deleted`

Could you please explain what is wrong with the approach.

Reply:
We need to raise the event and check if it was done without errors. 
If err != nil then we can update the finalizer to unlock it and let the CR be deleted.